lib/actions/create-invitation.ts

```diff
@@ -0,0 +1,65 @@
+'use server'
+
+import {Role} from '@prisma/client'
+import {z} from 'zod'
+import prisma from '~/prisma'
+import {getCurrentUser} from '~/utils/session'
+
+const schema = z.object({
+	email: z.string(),
+	role: z.nativeEnum(Role),
+	teamId: z.string()
+})
+
+// Create invitation to join a team
+export default async function createInvitation(
+	prevState: any,
+	formData: FormData
+) {
+	const user = await getCurrentUser()
+	if (!user)
+		return {
+			message: 'Unauthorized, no session',
+			type: 'error'
+		}
+
+	const parsed = schema.parse({
+		email: formData.get('email'),
+		role: formData.get('role'),
+		teamId: formData.get('teamId')
+	})
+
+	const membership = await prisma.membership.findFirst({
+		where: {teamId: parsed.teamId, userId: user.id}
+	})
+	if (!membership)
+		return {
+			message: 'Unauthorized, not a part of the team',
+			type: 'error'
+		}
+
+	try {
```

Comment:
Tiniest nitpick but maybe we should validate that the invitation is unique to the email + team?
<img width="748" alt="Screenshot 2024-02-05 at 10 10 10 PM" src="https://github.com/RubricLab/maige/assets/36115192/8fdeed90-8b10-4638-bae0-0d45107dbc7f">

Reply:
Great idea, added a uniqueness constraint for email + team ✅ 

Checking if user is already a part of team is hard because not all users will have the same email with their GitHub, so the email is a means to communicate the invitation rather than check for uniqueness, if that make sense. The only unique attribute for a user is their `userName` or `id` and we can't check for that here.